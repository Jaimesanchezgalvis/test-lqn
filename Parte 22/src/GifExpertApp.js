import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = ({ defaultCategories = ['One Punch'] }) => {
    const [categories, setCategories] = useState(defaultCategories)

    // const handleAdd = ()=>{
    //     setCategories(['HunterXhunter',...categories]);
    // }
    const hello = () => {
        console.log("hello")
    }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory
                setCategories={setCategories}
                nuevaprop={hello}
            />
            {/* linea de division */}
            <hr />
            <ol>
                {
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    )
}

export default GifExpertApp;
