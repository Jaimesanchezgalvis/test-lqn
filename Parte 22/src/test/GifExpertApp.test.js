import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas sobre <GifExpertApp/>', () => {

    test('hacer mach con el snapshot', () => {

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })

})
