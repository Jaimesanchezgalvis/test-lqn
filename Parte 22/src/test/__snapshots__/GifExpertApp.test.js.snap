// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifExpertApp/> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    nuevaprop={[Function]}
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas sobre <GifExpertApp/> hacer mach con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    nuevaprop={[Function]}
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
